src/handler.rs

```diff
@@ -1,124 +1,330 @@
+    let linked_count = from_other.len();
+    main_karma.aliases = Some(Aliases::FromOther(from_other));
+    if let Err(err) = main_karma.save(scope.clone(), dazeus) {
+        error!("error linking terms in main term: {}", err);
+        return;
+    }
+
+    // Log and reply to the user.
+    info!(
+        "{}karmalink command in {}/{} from '{}': linked {} terms to {}",
+        highlight_char, &evt[0], &evt[2], &evt[1], linked_count, &main_karma.term
+    );
+    let reply = format!(
+        "Linked {} terms together into {}",
+        linked_count, &main_karma.term
+    );
+    dazeus.reply(evt, &reply, false);
 }
 
-fn retrieve_all_karmas(evt: &Event, dazeus: &DaZeusClient) -> Vec<KarmaValue> {
-    let mut karmas = Vec::new();
-    for key in 5..evt.len() {
-        if !karmas.iter().any(|e: &KarmaValue| e.term == &evt[key]) {
-            karmas.push(match KarmaValue::from_dazeus(dazeus, Scope::network(&evt[0]), &evt[key]) {
-                Ok(karma) => karma,
-                _ => KarmaValue::new(&evt[key])
-            });
+pub fn reply_to_karmaunlink_command(evt: &Event, dazeus: &dyn DaZeusClient) {
+    let scope = Scope::network(&evt[0]);
+    let highlight_char = &dazeus.get_highlight_char().unwrap_or_default();
+    let term = &evt[5];
+
+    // Check if this term has a group that we can split.
+    let mut karma_group = KarmaGroup::get_from_dazeus_or_new(dazeus, scope.clone(), term);
+    if karma_group.karmas.is_empty() {
+        error!("new karma group should never be empty");
+        return;
+    }
+    if karma_group.karmas.len() == 1 {
+        let term = &karma_group.karmas.values().next().unwrap().term;
+        info!(
+            "{}karmaunlink command in {}/{} from '{}': {} is not linked",
+            highlight_char, &evt[0], &evt[2], &evt[1], term
+        );
+        let reply = format!("{} is not linked", &term);
+        dazeus.reply(evt, &reply, false);
+        return;
+    }
+
+    // Split the group.
+    for karma in karma_group.karmas.values_mut() {
+        karma.aliases = None;
+        if let Err(err) = karma.save(scope.clone(), dazeus) {
+            error!("error unlinking terms: {}", err);
+            return;
         }
     }
-    karmas
+
+    // Log and reply to the user.
+    let group_terms = karma_group
+        .karmas
+        .keys()
+        .map(|term| format!("'{}'", term))
+        .collect::<Vec<_>>()
+        .join(", ");
+    info!(
+        "{}karmaunlink command in {}/{} from '{}': unlinked {}",
+        highlight_char, &evt[0], &evt[2], &evt[1], group_terms
+    );
+    let reply = format!("Ack, unlinked {}", group_terms);
+    dazeus.reply(evt, &reply, false);
 }
+
+pub fn reply_with_redirect(to: &'static str, evt: &Event, dazeus: &dyn DaZeusClient) {
+    let msg = match dazeus.get_highlight_char() {
+        Some(highlight_char) => format!("Use '{}{}'", highlight_char, to),
+        None => format!("Use '{}' command", to),
+    };
+    dazeus.reply(&evt, &msg, true);
+}
```

Comment:
Has this been run through `rustfmt`? I don't think it allows no eol at EOF.

Reply:
I think I put this through `rustfmt`, but I'll double-check.